nomad/structs/config/tls.go

```diff
@@ -168,3 +182,22 @@ func (t *TLSConfig) Merge(b *TLSConfig) *TLSConfig {
 	}
 	return result
 }
+
+// Equals compares the fields of two TLS configuration objects, returning a
+// boolean indicating if they are the same.
+func (t *TLSConfig) Equals(newConfig *TLSConfig) bool {
+	if t == nil && newConfig == nil {
+		return true
+	}
+
+	if t != nil && newConfig == nil {
```

Comment:
Can newConfig be nil? Do we need the inverse of this check?

Reply:
newConfig _should_ not ever be nil, so I suppose we don't need to check for it? Not sure how careful we should be for errors in calling code.